pysass.cpp

```diff
@@ -404,6 +435,116 @@ static void _add_custom_functions(
     sass_option_set_c_functions(options, fn_list);
 }
 
+Sass_Import_List _call_py_importer_f(
+        const char* path,
+        Sass_Importer_Entry cb,
+        struct Sass_Compiler* comp
+) {
+    PyObject* pyfunc = (PyObject*)sass_importer_get_cookie(cb);
+    PyObject* py_path = PyUnicode_FromString(path);
+    PyObject* py_args = PyTuple_New(1);
+    PyObject* py_result = NULL;
+    Sass_Import_List sass_imports = NULL;
+    Py_ssize_t i;
+    
+    PyTuple_SetItem(py_args, 0, py_path);
+    
+    if (!(py_result = PyObject_CallObject(pyfunc, py_args))) {
+        sass_imports = sass_make_import_list(1);
+        sass_imports[0] = sass_make_import_entry(path, 0, 0);
+        
+        PyObject* exc = _exception_to_bytes();
+        char* err = PySass_Bytes_AS_STRING(exc);
+        
+        sass_import_set_error(sass_imports[0],
+                              err,
+                              0, 0);
+        
+        Py_XDECREF(exc);
+        Py_XDECREF(py_args);
+        Py_XDECREF(py_result);
+        return sass_imports;
+    }
+    
+    Py_XDECREF(py_args);
+        
+    if ( py_result == Py_None ) {
+        Py_XDECREF(py_result);
+        return 0;
+    }
+    
+    sass_imports = sass_make_import_list(PyList_Size(py_result));
+    
+    /* TODO: Iterator instead of literal list? Memory savings Python-side! */
+    for (i = 0; i < PyList_GET_SIZE(py_result); i += 1) {
+        PyObject* import_item = PyList_GET_ITEM(py_result, i);
+        char* path_str = NULL;  /* XXX: Memory leak? */
+        char* source_str = NULL;
+        char* sourcemap_str = NULL;
+        
+        /* GET_ITEM doesn't auto-retain. Future armoring relating to above
+           TODO, where iteration may return to the Python code context. */
+        Py_INCREF(import_item);
+        
+        /* TODO: Switch statement and error handling for default case. Better way? */
+        if ( PyTuple_GET_SIZE(import_item) == 1 ) {
+            fprintf(stderr, "il cb item 1-tup\n");
+            PyArg_ParseTuple(import_item, "es",
+                             0, &path_str);
+        } else if ( PyTuple_GET_SIZE(import_item) == 2 ) {
+            fprintf(stderr, "il cb item 2-tup\n");
+            PyArg_ParseTuple(import_item, "eses",
+                             0, &path_str, 0, &source_str);
+        } else if ( PyTuple_GET_SIZE(import_item) == 3 ) {
+            fprintf(stderr, "il cb item 3-tup\n");
+            PyArg_ParseTuple(import_item, "eseses",
+                             0, &path_str, 0, &source_str, 0, &sourcemap_str);
+        }
```

Comment:
Hmm, that's something I couldn't easily find.  What's the template for catching ParseTuple exceptions C-side?